### PR TITLE
feat: ensure that update only to the application signed with same cert

### LINKED ISSUE #1187: ensure that update only to the application signed with same cert
I want to be sure that Windows (NSIS) application can update only to the application signed with same cert. 

For Squirrel.Mac it works. When I try to update Mac application, that is not signed/signed with different cert to the correct application (signed with correct cert) I get error during update.

Same thing seems not working for Windows one.

@develar, could you please suggest something?
Thanks in advance.

### PR DESCRIPTION
**Pull request by MariaDima**
Close #1187

### CODE REVIEW

File: packages/electron-updater/src/NsisUpdater.ts [hunk 1]
```diff
@@ -58,14 +59,79 @@ export class NsisUpdater extends AppUpdater {
...
+    if (hasValidSignature) {
+      if (logger != null) {
+        logger.info(`New version ${this.versionInfo!.version} has been downloaded to ${tempFile}`)
+      }
+
+      this.setupPath = tempFile
+      this.addQuitHandler()
+      this.emit("update-downloaded", this.versionInfo)
+      return tempFile
+    } else {
```
Comment (develar) [inline packages/electron-updater/src/NsisUpdater.ts]: I think, we should emit error here.

File: packages/electron-updater/src/NsisUpdater.ts [hunk 2]
```diff
@@ -58,14 +59,79 @@ export class NsisUpdater extends AppUpdater {
...
+      }
+      return await remove(tempDir)
+    }
+  }
+
+  // $certificateInfo = (Get-AuthenticodeSignature 'xxx\yyy.exe'
+  // | where {$_.Status.Equals([System.Management.Automation.SignatureStatus]::Valid) -and $_.SignerCertificate.Subject.Contains("CN=circuitdev.siemens.com")})
+  // | Out-String ; if ($certificateInfo) { exit 0 } else { exit 1 }
+  private async spawnVerifySignature(tempUpdateFile: string) : Promise<boolean> {
+    const isWin = Platform.WINDOWS === Platform.fromString(process.platform)
```
Comment (develar) [inline packages/electron-updater/src/NsisUpdater.ts]: Platform not available in the electron-updater to reduce deps. Please use process.platform === "win32"

File: packages/electron-builder/src/winPackager.ts [hunk 3]
```diff
@@ -82,6 +82,10 @@ export class WinPackager extends PlatformPackager<WinBuildOptions> {
     return publisherName == null ? null : asArray(publisherName)
   })
 
+  get forceCodeSigningVerification(): boolean {
+    return (<WinBuildOptions>this.platformSpecificBuildOptions).forceCodeSigningVerification || false
```
Comment (develar) [inline packages/electron-builder/src/winPackager.ts]: Let's enable it by default:)

**Comment by develar** [comment]
Amazing. As new major version of electron-updater is published, it is a right time to make this check by default.

**Comment by develar** [comment]
Please do not forget to execute yarn lint.

**Comment by MariaDima** [comment]
Ok, updated to have the update signature validation enabled by default.
I left the option for someone to not perform the validation. Did you mean to remove the option completely?

What can we do about testing?
Is it possible to upload a signed executable under https://develar.s3.amazonaws.com/test/TestApp Setup 1.1.0.exe (or to another location)?

Thanks for looking into the PR!

**Comment by develar** [comment]
No option should be kept to not force users.

**Comment by develar** [comment]
I will resolve merge conflicts.

**Comment by develar** [comment]
Could you please `Allow edits from maintainers`?

**Comment by MariaDima** [comment]
"Allow edits from maintainers." is checked. Can't you edit?
image

**Comment by develar** [comment]
Merged — https://github.com/electron-userland/electron-builder/commit/66771d32e04ba1719cc0f0a81d480ddfdc9f237c 

Thanks a lot! It is amazing and must-have feature.

**Comment by MariaDima** [comment]
Thank you! Your enhancement rocks :-)

**Pull request closed without merging**